react-backend/client/src/components/visitors/qrcode.js

```diff
@@ -1,49 +1,55 @@
 /*global Instascan*/
 import React, { Component } from 'react';
 import { PurposeButton } from './purposeButton';
-import { withRouter} from 'react-router-dom';
-
+import { withRouter } from 'react-router-dom';
 
 function getUserFromQRScan(content) {
   return fetch('/getUsername', {
-    method: "POST",
+    method: 'POST',
     headers: {
-      'Content-Type': 'application/json'
+      'Content-Type': 'application/json',
     },
-    body: JSON.stringify({user:content})
-  })
-  .then(res=>res.json())
-  .catch((err) => {
-    console.log("error from getUserFromQRScan:  ", err);
-    throw err
+    body: JSON.stringify({ user: content }),
   })
+    .then(res => res.json())
+    .catch(err => {
+      console.log('error from getUserFromQRScan:  ', err);
+      throw err;
+    });
 }
 
 function instascan() {
-  return new Promise((resolve, reject)=>{
+  return new Promise((resolve, reject) => {
     let scanner = new Instascan.Scanner({
       video: document.getElementById('preview'),
-      scanPeriod: 5
+      scanPeriod: 5,
     });
     scanner.addListener('scan', function(content) {
       if (document.getElementById('preview')) {
-        document.getElementById('preview').pause();
-      }
-      resolve(content)
-    });
-    Instascan.Camera.getCameras().then((cameras) => {
-      if (cameras.length > 0) {
-        scanner.start(cameras[0]);
+        scanner
+          .stop()
+          .then(res => {
+            resolve(content);
+          })
+          .catch(error => {
+            reject('failure stopping scanner');
+          });
       } else {
-        console.error('No cameras found.');
+        document.getElementById('preview').play();
+        reject('ADD LISTENER FAILED');
       }
-    }).catch((err) => {
-      console.error(err);
     });
-  })
+    Instascan.Camera.getCameras()
+      .then(cameras => {
+        if (cameras.length > 0) {
+          scanner.start(cameras[0]);
+        } else {
+          reject('ERROR HAPPENING AT getCameras');
+        }
+      })
```

Comment:
You still need to have a `.catch` here, whenever a function returns a `Promise` it needs to have a `.then` and a `.catch`. Inside of the `.catch` here you could `reject` since this is all happening inside of a `Promise` you are writing. Then when you call your `instascan()` function, any failures inside of it will trigger the `.catch` you have on [line 111](https://github.com/FAC-11/DataPower/blob/b3eceabfdb304e1633a3711c884a93056be5aa26/react-backend/client/src/components/visitors/qrcode.js#L111)